From ce42824919a64c9e532331113a629acb2eedaea0 Mon Sep 17 00:00:00 2001
From: Jason Montleon <jmontleo@redhat.com>
Date: Fri, 6 Feb 2026 11:29:34 -0500
Subject: [PATCH 0006/1239] Revert "net: spacemit: Check for netif_carrier_ok()
 in emac_stats_update()"

This reverts commit fdb99161cbef29dffed56a1cbec13fce301167db.
---
 drivers/net/ethernet/spacemit/k1_emac.c | 34 +++++--------------------
 1 file changed, 7 insertions(+), 27 deletions(-)

diff --git a/drivers/net/ethernet/spacemit/k1_emac.c b/drivers/net/ethernet/spacemit/k1_emac.c
index 88e9424d2d51..220eb5ce7583 100644
--- a/drivers/net/ethernet/spacemit/k1_emac.c
+++ b/drivers/net/ethernet/spacemit/k1_emac.c
@@ -1099,13 +1099,7 @@ static int emac_read_stat_cnt(struct emac_priv *priv, u8 cnt, u32 *res,
 					100, 10000);
 
 	if (ret) {
-		/*
-		 * This could be caused by the PHY stopping its refclk even when
-		 * the link is up, for power saving. See also comments in
-		 * emac_stats_update().
-		 */
-		dev_err_ratelimited(&priv->ndev->dev,
-				    "Read stat timeout. PHY clock stopped?\n");
+		netdev_err(priv->ndev, "Read stat timeout\n");
 		return ret;
 	}
 
@@ -1153,25 +1147,17 @@ static void emac_stats_update(struct emac_priv *priv)
 
 	assert_spin_locked(&priv->stats_lock);
 
-	/*
-	 * We can't read statistics if the interface is not up. Also, some PHYs
-	 * stop their reference clocks for link down power saving, which also
-	 * causes reading statistics to time out. Don't update and don't
-	 * reschedule in these cases.
-	 */
-	if (!netif_running(priv->ndev) ||
-	    !netif_carrier_ok(priv->ndev) ||
-	    !netif_device_present(priv->ndev)) {
+	if (!netif_running(priv->ndev) || !netif_device_present(priv->ndev)) {
+		/* Not up, don't try to update */
 		return;
 	}
 
 	for (i = 0; i < sizeof(priv->tx_stats) / sizeof(*tx_stats); i++) {
 		/*
-		 * If reading stats times out anyway, the stat registers will be
-		 * stuck, and we can't really recover from that.
-		 *
-		 * Reading statistics also can't return an error, so just return
-		 * without updating and without rescheduling.
+		 * If reading stats times out, everything is broken and there's
+		 * nothing we can do. Reading statistics also can't return an
+		 * error, so just return without updating and without
+		 * rescheduling.
 		 */
 		if (emac_tx_read_stat_cnt(priv, i, &res))
 			return;
@@ -1650,12 +1636,6 @@ static void emac_adjust_link(struct net_device *dev)
 		emac_wr(priv, MAC_GLOBAL_CONTROL, ctrl);
 
 		emac_set_fc_autoneg(priv);
-
-		/*
-		 * Reschedule stats updates now that link is up. See comments in
-		 * emac_stats_update().
-		 */
-		mod_timer(&priv->stats_timer, jiffies);
 	}
 
 	phy_print_status(phydev);
-- 
2.54.0

